test/cpp/microbenchmarks/bm_fullstack_trickle.cc

```diff
@@ -204,9 +206,14 @@ class TrickledCHTTP2 : public EndpointPairFixture {
   gpr_timespec start_ = gpr_now(GPR_CLOCK_MONOTONIC);
 
   grpc_endpoint_pair MakeEndpoints(size_t kilobits) {
+    stats_ = grpc_passthru_endpoint_stats_create();  // is there a better way to
```

Comment:
I'm not really proud of the way stats_ gets set as a side effect of MakeEndpoints, but I wanted to try if this fixes the flake first. Any ideas from a C++ expert here?

Reply:
I'll send a sub-PR, but I'm speculating that you want something based on shared_ptr, make_shared, shared_from_this, etc

Reply:
As discussed offline, I didn't use shared_ptr as passthru_endpoint.cc is under test/core and other files there doesn't use shared_ptr there yet either.

Reply:
Can I suggest doing it in the constructor instead of in MakeEndpoints? This basically should only be created once either way, and it makes sense to create it in the constructor since it gets destroyed in the destructor.

Reply:
Not sure I understand what you mean.
I need to pass MakeEndpoints to the base class constructor and in the body of this class' constructor it's too late for doing that.
- grpc_endpoint_pair  is an "abstract" struct, so later extracting the stats from it (after EndpointPairFixture is constructed and knowing it's an passthru_endpoint) wouldn't work either.

Reply:
I think I solved it. PTAL.